services/business/checkers/virtual_services/route_checker.go

```diff
@@ -0,0 +1,97 @@
+package virtual_services
+
+import (
+	"reflect"
+	"strconv"
+
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/services/models"
+	"github.com/kiali/kiali/util/intutil"
+)
+
+type RouteChecker struct{ kubernetes.IstioObject }
+
+// Check returns both an array of IstioCheck and a boolean indicating if the current route rule is valid.
+// The array of IstioChecks contains the result of running the following validations:
+// 1. All weights with a numeric number.
+// 2. All weights have value between 0 and 100.
+// 3. Sum of all weights are 100 (if only one weight, then it assumes that is 100).
+// 4. All the route has to have weight label.
+func (route RouteChecker) Check() ([]*models.IstioCheck, bool) {
+	checks, valid := route.checkRoutesFor("http")
+	tcpChecks, tcpValid := route.checkRoutesFor("tcp")
+	return append(checks, tcpChecks...), valid && tcpValid
+}
+
+func (route RouteChecker) checkRoutesFor(kind string) ([]*models.IstioCheck, bool) {
+	validations := make([]*models.IstioCheck, 0)
+
+	var weightSum int
+	var weightCount int
+	var valid = true
+
+	http := route.GetSpec()[kind]
+	if http == nil {
+		return validations, valid
+	}
+
+	// Getting a []HTTPRoute
+	slice := reflect.ValueOf(http)
+	if slice.Kind() != reflect.Slice {
+		return validations, valid
+	}
+
+	for i := 0; i < slice.Len(); i++ {
```

Comment:
could you name "i" / "j" differently so that we easily understand what it is, like when reading `strconv.Itoa(i)` ?
Like `routeIdx` ?